### ISSUE TITLE
Restarting the Scheduler resets numberOfExecutionLeft

### ISSUE THREAD

**Issue by tobwiens** [ow2-proactive/scheduling#2518]
A task with "numberOfExecutionLeft":3
```
{"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":4,"parallelEnvironment":null,"name":"FailingTask","description":"\n\t\t\t\t This task fails \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":1460115708351,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":1460115701294,"executionDuration":-1,"numberOfExecutionLeft":4,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"FailingTask","id":0},"taskStatus":"IN_ERROR","executionHostName":"71b0edc6dc80 (local-LocalNodes)"},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"WaitingTask","description":"\n\t\t\t\t Waits for 4 seconds \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"WaitingTask","id":1},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"WaitingTask2","description":"\n\t\t\t\t Waits for 4 seconds \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"WaitingTask2","id":2},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"Python_Task","description":"\n\t\t\t\t The simplest task, ran by a python engine. \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"Python_Task","id":3},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}]>

```

Has "numberOfExecutionLeft":4

```
{"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":4,"parallelEnvironment":null,"name":"FailingTask","description":"\n\t\t\t\t This task fails \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":1460115708351,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":1460115701294,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"FailingTask","id":0},"taskStatus":"IN_ERROR","executionHostName":"71b0edc6dc80 (local-LocalNodes)"},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"WaitingTask","description":"\n\t\t\t\t Waits for 4 seconds \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"WaitingTask","id":1},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"WaitingTask2","description":"\n\t\t\t\t Waits for 4 seconds \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"WaitingTask2","id":2},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}, {"numberOfNodesNeeded":1,"iterationIndex":0,"replicationIndex":0,"maxNumberOfExecution":3,"parallelEnvironment":null,"name":"Python_Task","description":"\n\t\t\t\t The simplest task, ran by a python engine. \n\t\t\t","tag":null,"taskInfo":{"jobId":{"readableName":"script_python","id":1},"inErrorTime":-1,"scheduledTime":-1,"finishedTime":-1,"progress":0,"startTime":-1,"executionDuration":-1,"numberOfExecutionLeft":3,"numberOfExecutionOnFailureLeft":2,"taskId":{"readableName":"Python_Task","id":3},"taskStatus":"PAUSED","executionHostName":null},"maxNumberOfExecutionOnFailure":2,"genericInformations":{}}]>

```
, after the restart of the scheduler.

**Comment by lpellegr** [comment]
That's a behaviour we have also noticed with Marco. It is not something that changed recently, we were able to reproduce with 7.1.2.

Below is the link to the code which seems to be the cause of this behaviour:

https://github.com/ow2-proactive/scheduling/blob/master/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java#L417

**Comment by tobwiens** [comment]
The method createTaskData which you are referring to is not called when the scheduler is restarted. And I am focusing on the numberOfExecutionsLeft. 

Debugging the restart of the Scheduler, it seems that the numberOfExecutionsLeft is correctly written and read from the Database, but at some point that information gets lost.

**Comment by tobwiens** [comment]
Link: #2520

**Comment by tobwiens** [comment]
Solved by: #2520
